File: 75. 颜色分类/solution2.cpp
class Solution {  
public:  
    void sortColors(vector<int>& nums) {  
        int n = nums.size();  
        int p = 0;
        for(int i = 0; i < n; i++) {
            if(nums[i] == 0) {
                swap(nums[i], nums[p]);
                p++;
            }
        } 
        for(int i = p; i < n; i++) {
            if(nums[i] == 1) {
                swap(nums[i], nums[p]);
                p++;
            }
        } 
    }  
};

File: 数组/75. 颜色分类/solution2.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        p = 0
        n = len(nums)
        for i in range(n):
            if nums[i] == 0:
                nums[i], nums[p] = nums[p], nums[i]
                p += 1
        
        for i in range(p, n):
            if nums[i] == 1:
                nums[i], nums[p] = nums[p], nums[i]
                p += 1